src/lib/storage/storage_manager.cpp

```diff
@@ -7,45 +7,57 @@
 
 #include "utils/assert.hpp"
 
+#include <boost/range/combine.hpp>
+
 namespace opossum {
 
 StorageManager& StorageManager::get() {
-  return *(new StorageManager());
-  // A really hacky fix to get the tests to run - replace this with your implementation
+  static StorageManager instance;
+  return instance;
 }
 
 void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  _table_names.push_back(name);
+  _table_ptrs.push_back(table);
 }
 
 void StorageManager::drop_table(const std::string& name) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  auto table_index = get_table_index(name);
+  _table_names.erase(_table_names.begin() + table_index);
+  _table_ptrs.erase(_table_ptrs.begin() + table_index);
 }
 
 std::shared_ptr<Table> StorageManager::get_table(const std::string& name) const {
-  // Implementation goes here
-  return nullptr;
+  return _table_ptrs.at(get_table_index(name));
 }
 
 bool StorageManager::has_table(const std::string& name) const {
-  // Implementation goes here
-  return false;
+  return std::find(_table_names.begin(), _table_names.end(), name) != _table_names.end();
 }
 
-std::vector<std::string> StorageManager::table_names() const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
-}
+std::vector<std::string> StorageManager::table_names() const { return _table_names; }
 
 void StorageManager::print(std::ostream& out) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
-}
-
-void StorageManager::reset() {
-  // Implementation goes here
+  for (auto const& [table_name, table_ptr] : boost::combine(_table_names, _table_ptrs)) {
+    out << "=== " << table_name << " ===" << std::endl;
+    out << "n columns: " << table_ptr->column_count() << std::endl;
```

Comment:
Minor, use `\n` for newlines and only `std::endl` when you need to flush. Otherwise you flush unnecessarily often.